app/bundles/PluginBundle/Integration/AbstractIntegration.php

```diff
@@ -657,9 +657,16 @@ public function getDecryptedApiKeys($entity = false)
 
         $serialized = serialize($keys);
         if (empty($decryptedKeys[$serialized])) {
+            $decrypted = $this->decryptApiKeys($keys, true);
+            if (count($keys) !== 0 && count($decrypted) === 0) {
+                $decrypted = $this->decryptApiKeys($keys);
+                $this->encryptAndSetApiKeys($decrypted, $entity);
+                $this->em->persist($entity);
```

Comment:
Are you sure that $entity needs to be persisted? I think that it is already in database, isn't it?